### ISSUE TITLE
[API/Controller] Azure spec invalid

### ISSUE THREAD

**Issue by floreks** [kubermatic/kubermatic#5745]
It seems that the `AzureNodeSpec` zone field is not as in the Azure API. Our API and controller define it as an array of strings where it accepts a single string that defines zone number.

https://github.com/Azure/azure-quickstart-templates/blob/82a8b42aa327c92857b34f86f455aadbff4257cc/101-vm-simple-zones/azuredeploy.json#L90-L101

[Here](https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.management.network.models.publicipaddress?view=azure-dotnet) on the other hand `PublicIPAddress` accepts a list of zones. Which one we should be using?

If the list is acceptable then the frontend needs to allow selecting multiple zones also.

https://github.com/kubermatic/kubermatic/blob/0713c2a003812492d5e827e4f5d7b8956850e935/pkg/api/v1/types.go#L1117

The second issue is that providing `osDiskSize` and `dataDiskSize` seems to be ignored.

### Request
![Zrzut ekranu z 2020-08-10 12-11-24](https://user-images.githubusercontent.com/2285385/89772975-4b201500-db03-11ea-872e-f0918b882d1f.png)

### Response
![Zrzut ekranu z 2020-08-10 12-11-34](https://user-images.githubusercontent.com/2285385/89772980-4d826f00-db03-11ea-926c-eaf480b77318.png)

**Comment by zreigz** [comment]
cc @moadqassem

**Comment by moadqassem** [comment]
@shaase-ctrl This ticket here blocks another ticket in review, probably we need to move this to the next sprint in order to unblock any dependency.

**Comment by shaase-ctrl** [comment]
If we are not able to do both, yes we will move it.

**Comment by floreks** [comment]
After discussing with @moadqassem we agreed that the API spec is correct and UI should be adjusted.

/close
